cmd/kube-apiserver/app/server.go

```diff
@@ -279,6 +280,23 @@ func Run(s *options.APIServer) error {
 	}
 	storageDestinations.AddAPIGroup("", etcdStorage)
 
+	// autoscaling group
+	if !apiGroupVersionOverrides["autoscaling/v1"].Disable {
+		autoGroup, err := registered.Group(autoscaling.GroupName)
+		if err != nil {
+			glog.Fatalf("Autoscaling API is enabled in runtime config, but not enabled in the environment variable KUBE_API_VERSIONS. Error: %v", err)
+		}
+		if _, found := storageVersions[autoGroup.GroupVersion.Group]; !found {
+			glog.Fatalf("Couldn't find the storage version for group: %q in storageVersions: %v", autoGroup.GroupVersion.Group, storageVersions)
+		}
+		autoscalingEtcdStorage, err := newEtcd(s.EtcdServerList, api.Codecs, storageVersions[autoGroup.GroupVersion.Group], s.EtcdPathPrefix, s.EtcdQuorumRead)
```

Comment:
IIUC @lavalamp is working on this in https://github.com/lavalamp/kubernetes/commit/29cd25f690a2dc40b8daae240abbc1df6550f523